src/openrct2/windows/Intent.cpp

```diff
@@ -75,8 +75,8 @@ uint32 Intent::GetUIntExtra(uint32 key)
     }
 
     auto data = _Data.at(key);
-    openrct2_assert(data.type == IntentData::DT_UINT, "Actual type doesn't match requested type");
-    return data.uintVal;
+    openrct2_assert(data.type == IntentData::DT_INT, "Actual type doesn't match requested type");
```

Comment:
You assert on INT here, but request uint?

Reply:
I've combined UINT and SINT storage-wise, and get the proper type when fetching sint or uint

Reply:
Nevermind that comment…